File: android/domain/src/main/java/com/arakene/domain/responses/VideoListResponse.kt
package com.arakene.domain.responses

import kotlinx.serialization.SerialName


data class VideoListResponse(
    @SerialName("page") val page: Int,
    @SerialName("per_page") val perPage: Int,
    @SerialName("videos") val videos: List<VideoDto>,
    @SerialName("total_results") val totalResults: Int,
    @SerialName("next_page") val nextPage: String,
    @SerialName("url") val url: String
)

File: flutter/lib/responses/video_list_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'video_dto.dart';

part 'video_list_response.g.dart';

@JsonSerializable(explicitToJson: true)
class VideoListResponse {
  final int page;

  @JsonKey(name: 'per_page')
  final int perPage;

  final List<VideoDto> videos;

  @JsonKey(name: 'total_results')
  final int totalResults;

  @JsonKey(name: 'next_page')
  final String nextPage;

  final String url;

  VideoListResponse({
    required this.page,
    required this.perPage,
    required this.videos,
    required this.totalResults,
    required this.nextPage,
    required this.url,
  });

  factory VideoListResponse.fromJson(Map<String, dynamic> json) =>
      _$VideoListResponseFromJson(json);

  Map<String, dynamic> toJson() => _$VideoListResponseToJson(this);

  @override
  String toString() {
    return 'VideoListResponse(page: $page, perPage: $perPage, totalResults: $totalResults, nextPage: $nextPage, url: $url, videos: $videos)';
  }
}
